### ISSUE TITLE
Support for partitioned collection

### ISSUE THREAD

**Issue by hjalle** [imranmomin/Hangfire.AzureDocumentDB#36]
Since non-partitioned collections is deprecated, support for partitioning is now needed. 

> We lowered the minimum throughput for partitioned containers to 400 RU/s, removing the only drawback partitioned containers had.
> 
> We are planning to deprecate ability to create non-partitioned containers, as they do not allow you to scale elastically. If for some reason you still need a container without partition key, you can use our SDKs to create one. 

I made a local fork half a year ago, and if I recall correctly, it was only minor things that required changes to enable partitioning support. The most obvious is the error you get when using a partitioned collection (with partition "/type"):

> Upserts for scripts in collections with multiple partitions are not supported

Is this something that is planned?

**Comment by imranmomin** [comment]
No plans yet. But open to PR. Can you submit the PR

**Comment by erikpaperik** [comment]
@hjalle
I was facing this partition issue today while I was trying to connect Hangfire to a collection I defined upfront in the Azure Portal.

The only way to get Hangfire up and running was to let Hangfire do the setup of the DB/collection.

Nevertheless I ran into a problem when I tried to access the dashboard.
https://github.com/imranmomin/Hangfire.AzureDocumentDB/pull/35

**Comment by imranmomin** [comment]
@hjalle @erikpaperik I have merged the #35 pull-request. I have released the package on nuget

Please use the nuget package

```powershell
PM> Install-Package Hangfire.AzureDocumentDB -Version 3.0.1-develop-1
```

Let me know how it goes, so I can publish the final release 3.0.1

**Comment by imranmomin** [comment]
@hjalle @erikpaperik how did the testing go. 

Also should we pass `FeedOption` with `EnableCrossPartitionQuery` wherever I'm using `QueryText` instead of `LINQ`

**Comment by erikpaperik** [comment]
@imranmomin 

Thx for your fast response!
I'm not sure if I can test the `develop-1` NuGet this week but I'll definitly write you within the upcomming week.

Best
Erik

**Comment by erikpaperik** [comment]
@imranmomin 

I've just tried the `develop-1` NuGet Version in Azure and everything worked as expected!

Best regards
Erik

**Comment by imranmomin** [comment]
Awesome, so should also send the FeedOption wherever I'm using the RAW sql for query

**Comment by imranmomin** [comment]
@erikpaperik @hjalle thanks for reporting the issue. I have released the [3.0.1](https://www.nuget.org/packages/Hangfire.AzureDocumentDB/3.0.1)

Also, would you guys like to be collaborator for the project.
